templates/cluster-template-clusterclass.yaml

```diff
@@ -297,6 +297,31 @@ spec:
             - nutanix-quick-start-worker
     enabledIf: '{{if .additionalCategories}}true{{end}}'
     name: add-additional-categories
+  - definitions:
+    - jsonPatches:
+      - op: add
+        path: /spec/template/spec/gpus
+        valueFrom:
+          variable: gpus
+      selector:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+        kind: NutanixMachineTemplate
+        matchResources:
+          controlPlane: true
+    - jsonPatches:
+      - op: add
+        path: /spec/template/spec/gpus
+        valueFrom:
+          variable: gpus
+      selector:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+        kind: NutanixMachineTemplate
+        matchResources:
+          machineDeploymentClass:
```

Comment:
were we supporting assigning GPUs only for control plane nodes or to worker nodes? if yes then we need to separate out this?

Reply:
NutanixMachine has GPU fields, it can be mounted to any NutanixMachine regardless of whether it's CP or Worker.

Reply:
Made the changes for GPU. PTAL. AdditionalCategories should be handled in a separate PR IMO.